### PR TITLE
Make the point cloud colored

### PR DESCRIPTION
**Pull request by reunanen**
To address #149, keep color information related to point cloud points.

Mostly work by @vwvw – I'm just jumping in, because he's busy.

colored-point-cloud

### CODE REVIEW

File: src/openvslam/module/keyframe_inserter.cc [hunk 1]
```diff
@@ -137,7 +137,8 @@ data::keyframe* keyframe_inserter::insert_new_keyframe(data::frame& curr_frm) {
 
         // idxに対応する3次元がなければstereo triangulationで作る
         const Vec3_t pos_w = curr_frm.triangulate_stereo(idx);
-        auto lm = new data::landmark(pos_w, keyfrm, map_db_);
+        const auto color = curr_frm.get_keypoint_color(count);
```
Comment (reunanen) [inline src/openvslam/module/keyframe_inserter.cc]: @vwvw Here you had used `count` and not `idx`. Was that intentional? I didn't go into any depth in this regard.

File: src/pangolin_viewer/viewer.cc [hunk 2]
```diff
@@ -307,15 +309,17 @@ void viewer::draw_landmarks() {
         return;
     }
 
-    glPointSize(point_size_ * *menu_lm_size_);
-    glColor3fv(cs_.local_lm_.data());
+    glPointSize(point_size_ * *menu_lm_size_ * 2);
```
Comment (reunanen) [inline src/pangolin_viewer/viewer.cc]: Now that there's no distinction in color, make local landmarks larger.

File: src/openvslam/module/keyframe_inserter.cc (hunk 1 above)
Comment (vwvw) [reply src/openvslam/module/keyframe_inserter.cc]: No, probably a mistake! Good catch!

Comment (reunanen) [reply src/openvslam/module/keyframe_inserter.cc]: Changed to use `idx` and not `count`.

**Comment by shinsumicco** [comment]
@reunanen 
Sorry for the late reply, and thank you for the nice PullRequest. 
I have a question about processing time.
Is the consumed time for tracking of each frame increased after your implementation for this PullReq?
I'll check it ASAP, but we'd grateful if you could inspect it on your environment.
Thanks.

**Comment by reunanen** [comment]
@shinsumicco I can't see any practical difference in the time required for tracking:

* Before the change: 206 seconds
* After the change: 206 seconds

(All else equal. Using the `aist_living_lab_1` example video, `--frame-skip 1`, and the Pangolin viewer.)

I also profiled the CPU usage (using Visual Studio, not Google Perf Tools), and the lines changed in this PR were not really highlighted in any way.

**Comment by nickponline** [comment]
@reunanen  Excited for this PR to be merged. Thanks for working on it.

**Comment by nickponline** [comment]
I don't think the color appear correctly in the Socket viewer though.

**Comment by reunanen** [comment]
@nickponline You are right, the Socket viewer hasn't been touched. Want to contribute, if you are using it? Because I'm not.

**Comment by shinsumicco** [comment]
>     * Before the change: 206 seconds
>     * After the change: 206 seconds
@reunanen Thank you for the report.
Before a review, could you merge the latest master branch because the recent commit in google-test had caused a compile issue (#252) ?

**Comment by reunanen** [comment]
@shinsumicco Sure, merged.

**Pull request closed without merging**